components/tidb_query_expr/src/impl_string.rs

```diff
@@ -742,6 +742,14 @@ pub fn trim_1_arg(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write_ref(Some(result)))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn trim_2_args(arg: BytesRef, pat: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    let arg = String::from_utf8_lossy(arg);
```

Comment:
I think `Trim3Args` is buggy because of the usage of `from_utf8_lossy`. I think we should use [`bstr`](https://docs.rs/bstr/0.2.14/bstr/trait.ByteSlice.html#method.trim_start ).

Reply:
There is a test case for invalid utf8 for `Trim1Arg`. I'll port it to `Trim2Args` and `Trim2Args` tests. I've looked into TiDB tests, but they are quite terse and don't deal with non-utf8 characters. However, Go strings do not have to be valid utf-8, so `Trim3Args` is indeed buggy. I'll make the changes and ask for another review

Reply:
unfortunately, `bstr` does not support trimming by patterns longer than one characted. I ended up implementing it for byte slices and expanded the test suite